wren/service/materialization.go

```diff
@@ -181,6 +189,26 @@ func (s *materializationService) createMaterialization(ctx context.Context, owne
 	if err != nil {
 		return nil, err
 	}
+
+	sourceType := materialization.SourceTypeUnspecified
+	for _, table := range tableMap {
+		var newSourceType materialization.SourceType
+		switch table.Source.Source.(type) {
+		case *v1alpha.Source_Kaskada:
+			newSourceType = materialization.SourceTypeFiles
+		case *v1alpha.Source_Pulsar:
+			newSourceType = materialization.SourceTypeStreams
+		default:
+			log.Error().Msgf("unknown source type %T", table.Source.Source)
+			return nil, customerrors.NewInternalError("unknown table source type")
+		}
+		if sourceType == materialization.SourceTypeUnspecified {
+			sourceType = newSourceType
+		} else if sourceType != newSourceType {
+			return nil, customerrors.NewInvalidArgumentErrorWithCustomText("cannot materialize tables from different source types")
+		}
```

Comment:
Another good spot to remind ourselves to update once we can read directly from files

Reply:
lots of changes when we can read directly from files... would rather not leave todo's through the codebase for each changed required.